### PR TITLE
refactor: update `blas/ext/base/dnannsum` to follow current project conventions

### LINKED ISSUE #1478: [RFC]: refactor `blas/ext/base/dnannsum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dnannsum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dnannsum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AuenKr**
Resolves #1478 .

## Description

> What is the purpose of this pull request?

This pull request:

-   refactored `@stdlib/blas/ext/base/dnannsum` to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1478 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/benchmark/benchmark.js [hunk 1]
```diff
@@ -40,19 +42,16 @@ var dnannsum = require( './../lib/dnannsum.js' );
 */
 function createBenchmark( len ) {
 	var out;
-	var x;
-	var i;
+	var x = filledarrayBy( len, 'float64', clbk );
+	out = new Float64Array( 2 );
+	return benchmark;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/benchmark/benchmark.js]: Please cross check the conventions here, `x` shall be declared and assigned on separate lines.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -49,19 +51,16 @@ var opts = {
 */
 function createBenchmark( len ) {
 	var out;
-	var x;
-	var i;
+	var x = filledarrayBy( len, 'float64', clbk );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/benchmark/benchmark.native.js]: same comment here

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/benchmark/benchmark.ndarray.js [hunk 3]
```diff
@@ -40,19 +42,16 @@ var dnannsum = require( './../lib/ndarray.js' );
 */
 function createBenchmark( len ) {
 	var out;
-	var x;
-	var i;
+	var x = filledarrayBy( len, 'float64', clbk );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/benchmark/benchmark.ndarray.js]: same here

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/lib/ndarray.native.js [hunk 4]
```diff
@@ -40,28 +41,23 @@ var addon = require( './dnannsum.native.js' );
...
-	if ( strideOut < 0 ) {
-		offsetOut += strideOut;
-	}
-	viewX = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewOut = new Float64Array( out.buffer, out.byteOffset+(out.BYTES_PER_ELEMENT*offsetOut), out.length-offsetOut ); // eslint-disable-line max-len
+
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	offsetOut = minViewBufferIndex( N, strideOut, offsetOut );
+	viewX = offsetView( x, offsetX );
+	viewOut = offsetView( out, offsetOut );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/lib/ndarray.native.js]:
```suggestion
	offsetX = minViewBufferIndex( N, strideX, offsetX );
	offsetOut = minViewBufferIndex( N, strideOut, offsetOut );

	viewX = offsetView( x, offsetX );
	viewOut = offsetView( out, offsetOut );

```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Left a few comments, thanks for this PR!

**Comment by AuenKr** [comment]
@Pranavchiku applied all suggestion and fixed `addon.c` file which causing test case to fail.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/manifest.json [hunk 5]
```diff
@@ -1,42 +1,77 @@
...
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": []
+    },
+    {
+      "task": "examples",
+      "src": [
+        "./src/dnansum.c"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/manifest.json]:
```suggestion
        "./src/dnannsum.c"
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/manifest.json [hunk 6]
```diff
@@ -1,42 +1,77 @@
...
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-int64",
+        "@stdlib/napi/argv-strided-float64array"
+      ]
+    },
+    {
+      "task": "benchmark",
+      "src": [
+        "./src/dnansum.c"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/manifest.json]:
```suggestion
        "./src/dnannsum.c"
```

**Comment by AuenKr** [comment]
@Planeshifter can u review the PR?

**Review by Planeshifter — APPROVED** [review]
Thanks @AuenKr for your PR! Looked great, I just had to add a missing dependency to the `manifest.json` example and benchmarks configurations. Will merge shortly.

**Pull request merged**